### PR TITLE
docs: grammar and sentence structure fixes

### PR DESCRIPTION
**Pull request by Fdawgs**
Minor tweaks to the shorter documentation files to increase readability.
Plan on taking a pass at the bigger files at some point.

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Lifecycle.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Lifecycle
 Following the schema of the internal lifecycle of Fastify.<br>
-On the right branch of every section there is the next phase of the lifecycle, on the left branch there is the corresponding error code that will be generated if the parent throws an error *(note that all the errors are automatically handled by Fastify)*.
+On the right branch of every section, there is the next phase of the lifecycle, on the left branch there is the corresponding error code that will be generated if the parent throws an error *(note that all the errors are automatically handled by Fastify)*.
```
Comment (jsumners) [inline docs/Lifecycle.md:5]: This sentence doesn't make sense to me with the new comma. The two parts it splits seem to be one in my reading.

File: docs/Recommendations.md [hunk 2]
```diff
@@ -8,21 +8,20 @@ Fastify.
 * [Use A Reverse Proxy](#reverseproxy)
 * [Kubernetes](#kubernetes)
 
-## Use A Reverse Proxy
+## Use a Reverse Proxy
```
Comment (jsumners) [inline docs/Recommendations.md:11]: It seems to me this, and in the PR on errors, is an attempt to enforce https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case

My view is that we ignore those rules and either capitalize every word in a heading or capitalize only the first word. By trying to introduce any other syntax rules we just get into a mess like is described in https://www.chicagomanualofstyle.org/qanda/data/faq/topics/CapitalizationTitles/faq0007.html

My preference is to capitalize every word in a heading.

File: docs/Recommendations.md [hunk 3]
```diff
@@ -8,21 +8,20 @@ Fastify.
...
+with the language. With Node.js, one can write an application that
 _directly_ handles HTTP requests. As a result, the temptation is to write
 applications that handle requests for multiple domains, listen on multiple
-ports (i.e. HTTP _and_ HTTPS), and various other scenarios and combinations
-thereof. Further, the temptation is to then expose these applications directly
+ports (i.e. HTTP _and_ HTTPS), and then expose these applications directly
 to the Internet to handle requests.
 
-The Fastify team **strongly** considers this to be an anti-pattern and extremely
+The Fastify team **strongly** considers this an anti-pattern and extremely
```
Comment (jsumners) [inline docs/Recommendations.md:24]: Why?

File: docs/Recommendations.md (hunk 2 above)
Comment (Fdawgs) [reply docs/Recommendations.md:11]: There doesn't seem to be a consistent style for headings/titles across the documentation, i.e. the [Write-Plugin](https://github.com/fastify/fastify/blob/master/docs/Write-Plugin.md) file has a mix of lower and uppercase.

Happy to amend titles/headers to be consistent.

File: docs/Recommendations.md (hunk 3 above)
Comment (Fdawgs) [reply docs/Recommendations.md:24]: Sentence is still grammatically correct without "to be", and reduces the wordiness. 😬

File: docs/Lifecycle.md (hunk 1 above)
Comment (Fdawgs) [reply docs/Lifecycle.md:5]: On second glance you are correct, will remove!

File: docs/Recommendations.md (hunk 2 above)
Comment (jsumners) [reply docs/Recommendations.md:11]: Agreed that it isn't consistent across all of the documents. That is something we should address.

File: docs/Recommendations.md (hunk 3 above)
Comment (jsumners) [reply docs/Recommendations.md:24]: But does it have the same intention without "to be"?

Comment (Fdawgs) [reply docs/Recommendations.md:24]: After running this by my partner, who is an English teacher, she point blank said "to be" reads better.

...I expect to be single shortly.

Will revert!

**Review by jsumners — APPROVED** [review]
LGTM

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**